openmc/plotter.py

```diff
@@ -0,0 +1,617 @@
+                        funcs.append(nuc[mt].xs[nucT])
+                elif mt == UNITY_MT:
+                    funcs.append(lambda x: 1.)
+                elif mt == XI_MT:
+                    awr = nuc.atomic_weight_ratio
+                    alpha = ((awr - 1.) / (awr + 1.))**2
+                    xi = 1. + alpha * np.log(alpha) / (1. - alpha)
+                    funcs.append(lambda x: xi)
+                else:
+                    funcs.append(lambda x: 0.)
+            xs.append(openmc.data.Combination(funcs, op))
+    else:
+        raise ValueError(this.name + " not in library")
+
+    return energy_grid, xs
+
+
+def _calculate_xs_material(this, types, temperature=294., cross_sections=None):
+    """Calculates continuous-energy macroscopic cross sections of a
+    requested type
+
+    Parameters
+    ----------
+    this : openmc.Material
+        Material object to source data from
+    types : Iterable of values of PLOT_TYPES
+        The type of cross sections to calculate
+    temperature : float, optional
+        Temperature in Kelvin to plot. If not specified, a default
+        temperature of 294K will be plotted. Note that the nearest
+        temperature in the library for each nuclide will be used as opposed
+        to using any interpolation.
+    cross_sections : str, optional
+        Location of cross_sections.xml file. Default is None.
+
+    Returns
+    -------
+    energy_grid : numpy.array
+        Energies at which cross sections are calculated, in units of eV
+    data : numpy.ndarray
+        Macroscopic cross sections calculated at the energy grid described
+        by energy_grid
+
+    """
+
+    if this.temperature is not None:
+        T = this.temperature
+    else:
+        T = temperature
+
+    # Load the library
+    library = openmc.data.DataLibrary.from_xml(cross_sections)
+
+    # Expand elements in to nuclides with atomic densities
+    nuclides = this.get_nuclide_atom_densities(cross_sections)
+
+    # For ease of processing split out nuc and nuc_density
+    nuc_densities = [nuclide[1][1] for nuclide in nuclides.items()]
+
+    # Identify the nuclides which have S(a,b) data
+    sabs = {}
+    for nuclide in nuclides.items():
+        sabs[nuclide[0].name] = None
+    for sab_name in this._sab:
+        sab = openmc.data.ThermalScattering.from_hdf5(
+            library.get_by_material(sab_name)['path'])
+        for nuc in sab.nuclides:
+            sabs[nuc] = library.get_by_material(sab_name)['path']
+
+    # Now we can create the data sets to be plotted
+    xs = []
+    E = []
+    for nuclide in nuclides.items():
+        sab_tab = sabs[nuclide[0].name]
+        temp_E, temp_xs = calculate_xs(nuclide[0], types, T, sab_tab,
+                                       cross_sections)
+        E.append(temp_E)
+        xs.append(temp_xs)
+
+    # Condense the data for every nuclide
+    # First create a union energy grid
+    energy_grid = E[0]
+    for n in range(1, len(E)):
```

Comment:
Looks like I was comparing to a line that was close to 600, but there is another at around line 300. fixed now.